model/labels/float.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2025 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package labels
+
+import (
+	"bytes"
+	"math"
+	"strconv"
+	"sync"
+)
+
+// floatFormatBufPool is exclusively used in FormatOpenMetricsFloat.
+var floatFormatBufPool = sync.Pool{
+	New: func() any {
+		// To contain at most 17 digits and additional syntax for a float64.
+		b := make([]byte, 0, 24)
+		return &b
+	},
+}
+
+// FormatOpenMetricsFloat works like the usual Go string formatting of a float
+// but appends ".0" if the resulting number would otherwise contain neither a
+// "." nor an "e".
+func FormatOpenMetricsFloat(f float64) string {
```

Comment:
Does this placement makes sense? We also have another private method in expfmt in common: https://github.com/prometheus/common/blob/321962143e3c75b8a9df57564753bd568d5bba2b/expfmt/openmetrics_create.go#L679

cc @krajorama @bboreham